File: examples/ruby/pagination/sandbox.rb
# frozen_string_literal: true

require 'nightona'

nightona = Nightona::Nightona.new

nightona.list(Nightona::ListSandboxesQuery.new(
               limit: 10,
               labels: { 'env' => 'dev' },
               states: [Nightona::SandboxState::STARTED],
               sort: Nightona::SandboxListSortField::CREATED_AT,
               order: Nightona::SandboxListSortDirection::DESC
             )).each do |sandbox|
  puts sandbox.id
end

File: examples/python/pagination/sandbox.py
from nightona import Nightona, ListSandboxesQuery, SandboxListSortDirection, SandboxListSortField, SandboxState


def main():
    nightona = Nightona()

    for sandbox in nightona.list(
        ListSandboxesQuery(
            limit=10,
            labels={"env": "dev"},
            states=[SandboxState.STARTED],
            sort=SandboxListSortField.CREATEDAT,
            order=SandboxListSortDirection.DESC,
        )
    ):
        print(sandbox.id)


if __name__ == "__main__":
    main()
